docs/en/cookbook/dql-user-defined-functions.rst

```diff
@@ -3,69 +3,63 @@ DQL User Defined Functions
+* those that return a **string**
+* and those that return a **date.**
 
-The DQL Parser has hooks to register functions that can then be
-used in your DQL queries and transformed into SQL, allowing to
-extend Doctrines Query capabilities to the vendors strength. This
-post explains the User-Defined Functions API (UDF) of the Dql
-Parser and shows some examples to give you some hints how you would
-extend DQL.
-
-There are three types of functions in DQL, those that return a
-numerical value, those that return a string and those that return a
-Date. Your custom method has to be registered as either one of
+Your custom method has to be registered as either one of
 those. The return type information is used by the DQL parser to
-check possible syntax errors during the parsing process, for
-example using a string function return value in a math expression.
+check possible syntax errors during the parsing process (e.g.
+using a string function return value in a math expression).
 
 Registering your own DQL functions
 ----------------------------------
 
-You can register your functions adding them to the ORM
-configuration:
+You can register a function by adding it to the ORM configuration:
 
 .. code-block:: php
 
     <?php
-    $config = new \Doctrine\ORM\Configuration();
-    $config->addCustomStringFunction($name, $class);
-    $config->addCustomNumericFunction($name, $class);
-    $config->addCustomDatetimeFunction($name, $class);
+    use Doctrine\ORM\Configuration;
 
-    $em = new EntityManager($connection, $config);
+    $configuration = new Configuration();
+    $configuration->addCustomStringFunction($name, $class);
+    $configuration->addCustomNumericFunction($name, $class);
+    $configuration->addCustomDatetimeFunction($name, $class);
 
-The ``$name`` is the name the function will be referred to in the
-DQL query. ``$class`` is a string of a class-name which has to
-extend ``Doctrine\ORM\Query\Node\FunctionNode``. This is a class
-that offers all the necessary API and methods to implement a UDF.
+    $em = new EntityManager($connection, $config);
 
-Instead of providing the function class name, you can also provide
-a callable that returns the function object:
+* ``$name`` is the name of the function to be used in the DQL query.
+* ``$class`` is either a fully-qualified class name which has to
+extend ``Doctrine\ORM\Query\Node\FunctionNode``. Or a callable that returns the function object:
 
 .. code-block:: php
 
     <?php
-    $config = new \Doctrine\ORM\Configuration();
-    $config->addCustomStringFunction($name, function () {
+    use Doctrine\ORM\Configuration;
+
+    $configuration = new \Doctrine\ORM\Configuration();
+    $configuration->addCustomStringFunction($name, function() {
         return new MyCustomFunction();
     });
 
-In this post we will implement some MySql specific Date calculation
-methods, which are quite handy in my opinion:
+In this chapter we will implement some MySQL specific date calculation
```

Comment:
A cookbook doesn't consist in chapters, does it? I mean, it is not ordered.

Reply:
I never understood the separation of the docs into "Tutorials", "Reference", "Cookbook" - I just didn't take the time yet to think it through and come up with a better suggestion ;-)
Here I just wanted to get rid of the word "post" (2 occurrences), which just doesn't fit for docs, IMO. If you don't like "chapter", what about "article"?
I also thought about removing the author credit "Written by Benjamin Eberlei", since that doesn't fit for docs too IMO (all docs are a community effort with commits by various people). But since it's present throughout the cookbook, I thought I'll leave that for a separate PR too... ;-)

Reply:
Article seems fine 👍

Reply:
> I never understood the separation of the docs into "Tutorials", "Reference", "Cookbook" - I just didn't take the time yet to think it through and come up with a better suggestion ;-)

I think it's a bit like [the Diataxis framework](https://diataxis.fr/) if you equate Cookbook articles to how-to's